kvdb-web/src/indexed_db.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Utility functions to interact with IndexedDB browser API.
+
+use wasm_bindgen::{JsCast, JsValue, UnwrapThrowExt, closure::Closure};
+use web_sys::{
+	IdbDatabase, IdbRequest, IdbOpenDbRequest,
+	Event, IdbCursorWithValue,
+	IdbTransactionMode,
+};
+use js_sys::{Array, Uint8Array, ArrayBuffer};
+
+use futures::channel;
+use futures::prelude::*;
+
+use kvdb::{DBOp, DBTransaction};
+
+use std::ops::Deref;
+use log::{debug, warn};
+
+
+use crate::Column;
+
+
+/// Opens the IndexedDB with the given name and the specified number of columns
+/// (including the default one).
+pub fn open(name: &str, columns: u32) -> impl Future<Output = IdbDatabase> {
+	let (tx, rx) = channel::oneshot::channel::<IdbDatabase>();
+	// TODO: handle errors more gracefully,
```

Comment:
Let's handle errors in a better way before merging.
The easiest way to handle errors I think is to return `future::Either::B(future::err(...))`, and wrap the actual future into a `future::Either::A`.